### PR TITLE
#9 post new comment

### PR DESCRIPTION
**Pull request by DanielTaylor1998**
Hopefully everything is fine, I could probably make it clearer to user that a new comment is there ?

### CODE REVIEW

File: src/components/Commentform.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            setIsDisabled(true)
+
+            const body = {
+                username: loggedInUser.username,
+                body: text,
+            }
+
+            postComment(article_id, body)
+                .then((comment) => {
+                    console.log(comment, "new comment !")
```
Comment (Poonam-raj) [inline src/components/Commentform.jsx]: get rid of console.logs

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -13,19 +14,27 @@ export const Comments = (article_id) => {
...
         getComments(article_id.article_id)
             .then((comments) => {
-                setComments(comments)
+                setComments(comments.reverse())
                 setIsLoading(false)
-                console.log(comments)
             })
-    }, [])
+            .catch((err) => {
+                console.log(err)
```
Comment (Poonam-raj) [inline src/components/Comments.jsx]: get rid of console.logs

File: src/contexts/user.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { createContext, useState } from "react";
+
+export const UserContext = createContext();
+
+export const UserProvider = (props) => {
+    const [loggedInUser, setLoggedInUser] = useState({})
+    console.log(loggedInUser)
```
Comment (Poonam-raj) [inline src/contexts/user.js]: get rid of console logs

File: src/components/Users.jsx [hunk 4]
```diff
@@ -1,7 +1,46 @@
...
+                                <br />
+                                <button onClick={() => setLoggedInUser(user)}>Log In !</button>
+                            </div>
+                        </div>
+
+                    )
+                })}
+            </div>
+
+            <div className="col"></div>
```
Comment (Poonam-raj) [inline src/components/Users.jsx:51]: is this div needed?

File: src/components/Users.jsx [hunk 5]
```diff
@@ -1,7 +1,46 @@
...
+                                <h1>{user.username}</h1>
+                                <img className="profile-pic" src={user.avatar_url} />
+                                <br />
+                                <button onClick={() => setLoggedInUser(user)}>Log In !</button>
+                            </div>
+                        </div>
+
+                    )
+                })}
+            </div>
```
Comment (Poonam-raj) [inline src/components/Users.jsx:49]: A lot of divs going into the code- see if there are any semantic elements you could insert into the code

File: src/components/Users.jsx [hunk 6]
```diff
@@ -1,7 +1,46 @@
...
+            <div className="col"></div>
+            <div className="content">
+                {users.map((user) => {
+                    return (
+                        <div className="user-list" key={user.name}>
+                            <div className="userCard">
+                                <h1>{user.username}</h1>
+                                <img className="profile-pic" src={user.avatar_url} />
+                                <br />
+                                <button onClick={() => setLoggedInUser(user)}>Log In !</button>
```
Comment (Poonam-raj) [inline src/components/Users.jsx]: WHen I click on a button to log in I get no feedback that the log in has been successful, please make sure we either redirect to a profile page / there is a message saying I'm logged in / anything else to that effect

File: src/components/Commentform.jsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    return (
+        <div>
+            <div hidden={isHidden}>
+                <label>
+                    Please enter you comment:
+                    <br />
+                    <textarea className="textBox" onChange={handleChange}></textarea>
+                    <br />
+                    <button disabled={isDisabled} onClick={() => { comment(article_id) }}>Sumbit</button>
```
Comment (Poonam-raj) [inline src/components/Commentform.jsx]: Submit is spelt wrong

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Few things I want you to amend, send another PR our way when done 🚀

**Comment by k-huggs** [comment]
When you log in, you ned to provide some indication that it has been successful, potentially navigate to a new page or send an alert that it's been successful. - also provide some indication as to who is logged in

Clear the comments box once a comment has been posted

**Comment by k-huggs** [comment]
Nice all good to merge!

**Pull request merged**